wavelink/queue.py

```diff
@@ -155,37 +78,122 @@ class Queue(_Queue):
 
             Check whether a specific track is in the queue.
 
+        .. describe:: queue[1] = track
+
+            Set a track in the queue at a specific index.
+
+        .. describe:: del queue[1]
+
+            Delete a track from the queue at a specific index.
+
+        .. describe:: reversed(queue)
+
+            Return a reversed iterator of the queue.
 
     Attributes
     ----------
     history: :class:`wavelink.Queue`
-        A queue of tracks that have been added to history.
-
-        Even though the history queue is the same class as this Queue some differences apply.
-        Mainly you can not set the ``mode``.
+        A queue of tracks that have been added to history. Tracks are added to history when they are played.
     """
 
-    def __init__(self, history: bool = True) -> None:
-        super().__init__()
-        self.history: Queue | None = None
-
-        if history:
-            self.history = Queue(history=False)
+    def __init__(self, *, history: bool = True) -> None:
+        self._items: list[Playable] = []
 
-        self._loaded: Playable | None = None
+        self._history: Queue | None = None if not history else Queue(history=False)
```

Comment:
You can swap this so it removes negation
`self._history: Queue | None = Queue(history=False) if history else None`